.github/scripts/archive_eol.py

```diff
@@ -0,0 +1,108 @@
+import os
+import subprocess
+
+import github
+from github.GithubException import RateLimitExceededException, UnknownObjectException
+
+
+def ignore_ref(ref: str) -> bool:
+    ref_splits = ref.split("/")
+
+    if len(ref_splits) != 4:
+        return True
+
+    if ref_splits[2] not in ("x86_64", "aarch64") or ref_splits[1].endswith(
+        (".Debug", ".Locale", ".Sources")
+    ):
+        return True
+    return False
+
+
+def get_eol_refs(arch: str, remote: str) -> set:
+    cmd = (
+        f'flatpak remote-ls --user --arch="{arch}" --all --columns=ref,options {remote}'
+    )
+    ret = subprocess.run(cmd, shell=True, capture_output=True, encoding="utf-8")
+    eol_refs: set = set()
+
+    if ret.returncode == 0 and ret.stdout:
+        for line in ret.stdout.split("\n"):
+            if line:
+                split = line.split("\t")
+            if len(split) < 2:
+                continue
+            ref, options = split
+            if ignore_ref(ref):
+                continue
+            refname = ref.split("/")[1]
+            if any(x in options for x in ("eol=", "eol-rebase=")):
+                eol_refs.add(refname)
+
+    return eol_refs
+
+
+def main() -> None:
+
+    token = os.environ["GITHUB_TOKEN"]
+
+    g = github.Github(auth=github.Auth.Token(token))
+
+    # Exclude refs that have EOL notices but are still maintained
+    # in some branch
+    excludes = {
+        "com.github.eneshecan.WhatsAppForLinux",
```

Comment:
Something needs to be done here,, not sure what.

If we restrict this to `app/` only we will miss extensions, and even with `app/` only we will still need exclusions.


Maybe we can do a initial run to get all the archived ones, then set up some commit activity limit of 1 year or more. 

If it is appearing in EOL list and no activity in 1+ year it is a good bet that it is actually EOL.

Reply:
Last commit got rid of all false positives, but I kept the list around. It also got rid of some genuine repos that needed archiving but we can get them later.

This is the state now.

<details>

<summary> List </summary>

```
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Latte-Grey. Repo is in EOL list. Last push: 2022-12-13T09:51:38+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/io.github.vegad.vnotes. Repo is in EOL list. Last push: 2023-06-06T07:56:34+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/io.github.jazztickets.openflap. Repo is in EOL list. Last push: 2022-10-10T14:45:54+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.freedesktop.Sdk.Extension.gfortran-62. Repo is in EOL list. Last push: 2018-06-19T20:55:15+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.freedesktop.Sdk.Extension.mono5. Repo is in EOL list. Last push: 2022-04-25T00:56:04+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/com.bitstower.Markets. Repo is in EOL list. Last push: 2023-03-03T16:47:02+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/io.gitlab.JakobDev.jdSimpleAutostart. Repo is in EOL list. Last push: 2023-05-08T12:06:38+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Latte-Mauve. Repo is in EOL list. Last push: 2022-12-05T00:02:43+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Latte-Peach. Repo is in EOL list. Last push: 2022-12-13T10:01:21+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Breeze-Dark. Repo is in EOL list. Last push: 2022-02-22T00:19:33+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Mocha-Teal. Repo is in EOL list. Last push: 2022-12-13T09:51:10+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Latte-Teal. Repo is in EOL list. Last push: 2022-12-13T09:50:17+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Mocha-Pink. Repo is in EOL list. Last push: 2022-12-13T09:51:21+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gnome.Dictionary. Repo is in EOL list. Last push: 2023-04-16T18:19:01+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/io.github.tofudd.lanchat.terminal. Repo is in EOL list. Last push: 2023-03-25T17:32:01+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Mocha-Grey. Repo is in EOL list. Last push: 2022-12-13T09:51:16+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Mocha-Yellow. Repo is in EOL list. Last push: 2022-12-13T09:58:37+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Mocha-Lavender. Repo is in EOL list. Last push: 2022-12-05T00:02:39+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/uk.co.ibboard.cawbird. Repo is in EOL list. Last push: 2023-02-18T15:23:19+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.kde.PlatformInputContexts.MaliitSailfishOS. Repo is in EOL list. Last push: 2023-02-12T18:31:51+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/space.crankshaft.Crankshaft. Repo is in EOL list. Last push: 2023-05-25T20:45:27+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Mocha-Red. Repo is in EOL list. Last push: 2022-12-13T09:50:42+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/io.github.fkinoshita.Wildcard. Repo is in EOL list. Last push: 2023-06-28T21:21:01+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gnome.Crosswords.PuzzleSets.puzzlepull. Repo is in EOL list. Last push: 2023-01-04T02:04:17+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Mocha. Repo is in EOL list. Last push: 2022-12-13T09:50:11+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Latte. Repo is in EOL list. Last push: 2022-12-13T09:52:35+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.freedesktop.Sdk.Extension.node10. Repo is in EOL list. Last push: 2021-09-08T09:42:17+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Latte-Yellow. Repo is in EOL list. Last push: 2023-02-04T17:19:50+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/cz.pervoj.valdo-gtk. Repo is in EOL list. Last push: 2022-06-16T16:02:35+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Latte-Lavender. Repo is in EOL list. Last push: 2022-12-05T00:02:37+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/io.github.jazztickets.irrlamb. Repo is in EOL list. Last push: 2022-10-11T19:25:16+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Mocha-Mauve. Repo is in EOL list. Last push: 2022-12-05T00:02:41+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Latte-Green. Repo is in EOL list. Last push: 2022-12-13T10:01:21+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Mocha-Green. Repo is in EOL list. Last push: 2022-12-13T09:51:32+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Latte-Red. Repo is in EOL list. Last push: 2022-12-13T09:57:36+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/org.gtk.Gtk3theme.Catppuccin-Latte-Pink. Repo is in EOL list. Last push: 2022-12-13T09:58:02+00:00, earlier than: 2023-07-01T00:00:00+00:00
Archiving: https://github.com/flathub/br.gov.fazenda.receita.irpf. Repo is in EOL list. Last push: 2023-05-04T23:12:42+00:00, earlier than: 2023-07-01T00:00:00+00:00
```

</details>